compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -1316,11 +1442,10 @@ pub fn expand_preparsed_format_args(
         }
 
         diag.emit();
-    } else if cx.invalid_refs.is_empty() && !named_arguments.is_empty() {
+    } else if cx.invalid_refs.is_empty() && cx.ecx.sess.err_count() == 0 {
```

Comment:
Don't we suppress lints if we have errors? Can you describe what case(s) this prevents erroneous lints from being fired?